components/dropdown/dropdown-content-mixin.js

```diff
@@ -461,7 +466,8 @@ export const DropdownContentMixin = superclass => class extends RtlMixin(supercl
 			});
 
 			const spaceRequired = {
-				height: contentRect.height + headerFooterHeight + 10,
+				height: this.maxHeight !== null && contentRect.height > this.maxHeight
```

Comment:
Would you not want to check? 
```
contentRect.height + headerFooterHeight + 10 > this.maxHeight
```

Reply:
The +10 is padding related and needs to be added to the result either way, but good catch on the header and footer part I fixed that :)